icechunk/src/repository.rs

```diff
@@ -909,6 +914,22 @@ impl From<Repository> for ChangeSet {
     }
 }
 
+pub fn is_prefix_match(key: &str, prefix: &str) -> bool {
+    let tomatch =
+        if prefix != &String::from('/') { key.strip_prefix(prefix) } else { Some(key) };
```

Comment:
Can you avoid the allocation by saying `prefix != "/"` ?

Reply:
This code is tricky and somewhat hard to follow. I wonder, can we use `Path` methods for this? Path's should be able to tell you if they are parent/child right?

Reply:
That would also skip the somewhat expensive conversion from Path to String